### PR TITLE
Refine use of aria-controls on mobile masthead, fixes #655

### LINKED ISSUE #655: Masthead: Remove unnecessary aria-control from standalone mobile nav items
## Proposal
On the mobile menu of masthead, `aria-controls` is on unnecessary links. We need to remove unnecessary aria controls in navigation that isn't an accordion because aria-controls denotes that it's controlling something.

screen shot 2018-10-08 at 4 06 37 pm

Further reading on aria controls:
https://thesocietea.org/2016/09/aria-roles-and-attributes-how-to-actually-use-them/
image

## Type of change
Please select only 1.
* [ ] New Component
* [x] Enhancement to existing component
* [ ] Substantial documentation change
* [ ] Changes to the design or features of [sparkdesignsystem.com](http://sparkdesignsystem.com)

## Change category
Select as many as apply.
* [ ] **Visual** (Changes to the visual design of a component)
* [x] **UX** (Change the user experience design of a component)
* [x] **Component** (Modifying the functionality of a component)
* [x] **Editorial** (Changes to documentation or content style)
* [ ] **Code Style** (Changes to the way code is written or the component architecture)
* [ ] **Accessibility** (Substantial changes to a component's accessibility)

### PR DESCRIPTION
**Pull request by vjwilson**
I also noticed that the 2 links that were being used as triggers for accordions were pointing to the same non-existent ID, so I touched those up, too, as being directly related to the issue at hand.

Commit:
Refine use of aria-controls in narrow navigation example on mobile masthead pattern

### CODE REVIEW

**Comment by RCopeland** [comment]
Hey @vjwilson, thanks for picking this up!

File: src/patterns/components/navigation/narrow.hbs [hunk 1]
```diff
@@ -34,15 +34,15 @@ hasAngularCodeInfo: true
...
+      <a class="sprk-c-Accordion__summary {{summary}}" href="#">
         <span class="sprk-b-TypeBodyTwo sprk-c-Accordion__heading {{heading}}">
           Item 2
         </span>
       </a>
     </li>
 
     <li class="sprk-c-Accordion__item {{item}}" data-sprk-toggle="container">
-      <a aria-controls="details1" class="sprk-c-Accordion__summary {{summary}}" data-sprk-toggle="trigger" data-sprk-toggle-type="accordion" href="#">
+      <a aria-controls="details3" class="sprk-c-Accordion__summary {{summary}}" data-sprk-toggle="trigger" data-sprk-toggle-type="accordion" href="#">
```
Comment (RCopeland) [inline src/patterns/components/navigation/narrow.hbs]: I see you're removing the aria-controls from one of the accordions and updating the ids for the others. 
I think we want to just remove the use of aria-controls on all the accordions.

Comment (vjwilson) [reply src/patterns/components/navigation/narrow.hbs]: @RCopeland 

That's what I thought when I first read it at first, but when I dug into the linked doc, and saw that the note on the screenshot was on the middle link--the only one that didn't open an accordion--I changed my mind about the ask.

And if we do still need `aria-controls` on items 1 and 3, then they would need those unique IDs.

@RVMendoza , could you weight in on whether my PR is what you wanted?

Comment (RVMendoza) [reply src/patterns/components/navigation/narrow.hbs]: @vjwilson this is it! It makes sense that the aria controls match the the corresponding IDs, thanks for the extra effort.

**Comment by vjwilson** [comment]
@RVMendoza 

I went to make the changes in the master branch, but there's a bug in the master branch that not allowing me to check it by opening the nav when it's in mobile view.

It doesn't throw any errors itself, but there are two errors in the console whenever you first load the styleguide that aren't there when I run the staging branch. I even re-cloned the repo fresh, and the errors are there and the menu doesn't open before I make any changes. 

```
long.js?595c:15 Uncaught Error: Cannot find module "../../../../../packages/spark-core/es5/getElements"
    at webpackMissingModule (long.js?595c:15)
    at Object.defineProperty.value (long.js?595c:15)
    at __webpack_require__ (bootstrap 26178fc597cb61b4b51c?476e:19)
    at Object._self (drizzle.js?2ed9:6)
    at __webpack_require__ (bootstrap 26178fc597cb61b4b51c?476e:19)
    at bootstrap 26178fc597cb61b4b51c?476e:39
    at bootstrap 26178fc597cb61b4b51c?476e:39
webpackMissingModule @ long.js?595c:15
Object.defineProperty.value @ long.js?595c:15
__webpack_require__ @ bootstrap 26178fc597cb61b4b51c?476e:19
_self @ drizzle.js?2ed9:6
__webpack_require__ @ bootstrap 26178fc597cb61b4b51c?476e:19
(anonymous) @ bootstrap 26178fc597cb61b4b51c?476e:39
(anonymous) @ bootstrap 26178fc597cb61b4b51c?476e:39
toolkit.js:1 Uncaught Error: Cannot find module "../../../../packages/spark-core/es5/sparkCore"
    at webpackMissingModule (toolkit.js:1)
    at Object.<anonymous> (toolkit.js:1)
    at __webpack_require__ (bootstrap 26178fc597cb61b4b51c?476e*:19)
    at bootstrap 26178fc597cb61b4b51c?476e*:39
    at bootstrap 26178fc597cb61b4b51c?476e*:39
```

I'd like to help track down this regression, but I'm not familiar with Spark (or Drizzle), so any help or guidance would be appreciated.

(I could PR my changes into master as it is, but I don't like the idea of muddying code that not currently working.)

**Comment by yodasw16** [comment]
I don’t have my computer to confirm this bug until tomorrow morning, so I would log that as a separate bug, if you don’t mind. Thanks for digging in!

**Comment by RCopeland** [comment]
@vjwilson thanks for digging in! I think what's going on is that we made a fix in staging for this that we didn't make in master. It involves including the es5 transpile as part of the local build process. I have a PR open #668, that should fix it hopefully.

**Comment by vjwilson** [comment]
@RCopeland and @RVMendoza ,

What's the best course of action for this PR?
- should I leave it pointed at staging, so that it eventually gets into master with the es5 fix?
_or_
- should I open a PR with the same changes into master, even though you want be able to see them yet?

**Comment by RCopeland** [comment]
@vjwilson I just merged a fix into master for the build issues. Once you rebase on master, run npm run build-spark (hopefully that will take care of the es5 errors you're seeing) and then you can open your PR against master. Thanks!

**Pull request merged**